### DISCUSSION TITLE
Multisegment 1 LED Not Lighting

### DISCUSSION THREAD

**Discussion by spikked27** [awawa-dev/HyperSPI#30] [General]
Hello,
I've successfully set up a multi segment HyperSPI  instance with 634 total LEDs and man is it fast!  The system works exactly as I'd expect except that the last LED on the end of the 2nd segment doesn't light up, the segments have an equal amount of LEDs, 317 on one and 317 on the other so -DSECOND_SEGMENT_START_INDEX=317 and -DSECOND_SEGMENT_REVERSED, HyperHDR is similarly set up with, 206 on the Top/bottom and 111 on each side but still that last LED doesn't light up.  If I add an additional LED in the HyperHDR configuration to that side it will light up so I know if works, just not sure if I'm configuring either HyperSPI wrong or HyperHDR, any help is appeciated!

![Screenshot 2023-08-03 124249](https://github.com/awawa-dev/HyperSPI/assets/26699225/455d1f9b-e402-40b1-af67-fec8ecc935b4)
![20230803_124154](https://github.com/awawa-dev/HyperSPI/assets/26699225/e7aeb671-d2bc-446d-8491-9252569df629)
![20230803_124202](https://github.com/awawa-dev/HyperSPI/assets/26699225/51efce51-3aee-49fe-a87e-58be1fbb5376)

**Comment by awawa-dev** [comment]
Hi
Are you using the compiled firmware from your fork of the HyperSPI project?
I think that could be a problem: -DSECOND_SEGMENT_START_INDEX=318  instead of 317
So for -DSECOND_SEGMENT_START_INDEX=317:
First segment is [0..316] => 317 leds
Second segment is [317..633] => 317 leds
634 LEDs in HyperHDR configuration.

```
from your commit https://github.com/awawa-dev/HyperSPI/commit/af57fe9910eda2820e211b84e5ec1242286da772
[env:s2_mini_SK6812_RGBW_COLD]
build_flags = -DNEOPIXEL_RGBW -DCOLD_WHITE -DDATA_PIN=2 ${env.build_flags}
build_flags = -DNEOPIXEL_RGBW -DCOLD_WHITE -DDATA_PIN=33 -DSECOND_SEGMENT_START_INDEX=318 -DSECOND_SEGMENT_DATA_PIN=34 -DSECOND_SEGMENT_REVERSED 
```

**Reply by spikked27** [reply]
You are 100000% correct, I made the change to 317 in the ESP32 section and not in the ESP32-S2 section, please excuse my stupidity and thank you for looking into it.  Now to test the limits of this RPi4....
